src/search.F90

```diff
@@ -73,7 +74,8 @@ function binary_search_int4(array, n, val) result(array_index)
     R = n
 
     if (val < array(L) .or. val > array(R)) then
-      call fatal_error("Value outside of array during binary search")
+      array_index = -1
```

Comment:
And here. If you are pushing for a move towards the use of more error codes (and requiring the caller to handle error checking) then I think the potential error codes should be defined in the comment block above the function.